### PR TITLE
Change to store sampling rates in frame headers.

### LINKED ISSUE #195: Encoded flacs seem to only play about half way before halting. They don't play at all using QT Player.
Either using my custom implementation of flacenc or the provided CLI, the resulting flac file only plays about half way (when testing through chrome) and when testing in Quicktime Player no audio plays period (but the duration is correct).

(Of course, I tested other flac files from samples online, they all play as expected.)

Purely testing against PCM16 wavs at the moment.

Here is the source / sample:

[broken.zip](https://github.com/user-attachments/files/16449158/broken.zip)

### PR DESCRIPTION
**Pull request by yotarok**
This closes #195.

### CODE REVIEW

File: src/coding.rs [hunk 1]
```diff
@@ -434,6 +434,8 @@ fn encode_frame_impl(
 
     // for Claxon compatibility.
     frame.header_mut().reset_sample_size_spec(stream_info);
+    // for QuickTime compatibility.
+    frame.header_mut().reset_sample_rate_spec(stream_info);
```
Comment (dtzxporter) [inline src/coding.rs:438]: Is calling this twice redundant?

**Comment by dtzxporter** [comment]
With flac --test:

image

**Comment by dtzxporter** [comment]
Looks like a problem with 'custom' sample rates, the ones that work have one of the stock ones.

**Comment by yotarok** [comment]
Thanks for testing. I found a mistake in the previous commit `009f63d`. The extra sampling-rate bits (only used when sampling rates are atypical) are written in a wrong bitstream. The integration test is improved to cover this.

**Comment by dtzxporter** [comment]
Awesome. This fixes the issue playing the encoded flacs in QT/Chrome perfectly.

It does not solve the issue with the last block's samples being included in the output of any decoder, which also causes checksums to be incorrect in most tools.

With libflac, it will decode fine, but the resulting wav will have a bunch of extra empty samples at the end (however, it truncates the data size itself, it's just writing bad data...

With ffmpeg/fq (a cool diagnostic tool) they calculate the checksum wrong (because of the extra samples ~3000 in the example I gave above) and spit out the wrong sample count as well.

FQ output:
image

**Comment by yotarok** [comment]
> It does not solve the issue with the last block's samples being included in the output of any decoder, which also causes checksums to be incorrect in most tools.

I'm aware of this issue. I was thinking that it's not problematic to have extra zero-samples on the end of the file, as it is truncated in flac reference decoder.
But, since ffmpeg warns about this, it might be an actual issue that needs to be fixed.

I was reluctant to implement the last-frame truncation (as in flac reference encoder) because `flacenc-rs` states that it currently only supports encoding with "fixed" block-size, and the last-frame truncation violates this statement.
I'm now thinking implementing it, but anyway this should be in another PR.

**Pull request merged**